force-app/main/default/classes/TreeViewRecords.cls

```diff
@@ -0,0 +1,37 @@
+public with sharing class TreeViewRecords {
+    @AuraEnabled(cacheable=true)
+    public static List<TreeViewWrapper> getRecordsTreeView(String objectAPITreeView, String fieldsAPI){
+        
+        String query = '';
+        String[] fields;
```

Comment:
You can initialize the array from the beginning, in this way you don't need to check 
if(fields.size() > 0) 
also instead checking if size > 0, is better to use !fields,isEmpty()

Reply:
In this case, I've declared fields to be visible in the context.
If there is no fieldsAPI declared, the array will not be populated.
But I've changed the check with !array.empty() method.